randovania/generator/hint_distributor.py

```diff
@@ -510,21 +522,27 @@ def get_hint_precision(
 
     def add_hints_precision(
         self,
-        player_state: PlayerState,
         patches: GamePatches,
         rng: Random,
         player_pools: Sequence[PlayerPool],
+        hint_kinds: Container[HintNodeKind] = set(HintNodeKind),
     ) -> GamePatches:
         """
         Adds precision to all hints that are missing one.
         :param player_state:
         :param patches:
         :param rng:
+        :param player_pools:
+        :param hint_kinds: Which `HintNodeKind`s should be replaced?
```

Comment:
This is still weird. The function is all about touching the `Hint` objects, but these don't have a `HintNodeKind`.

Reply:
not really? the function is all about touching `patches.hints`, which is a `dict[NodeIdentifier, Hint]`. both key and value are important here